pallets/delegation/src/lib.rs

```diff
@@ -87,47 +88,87 @@ pub mod pallet {
 	pub struct Pallet<T>(_);
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			migrations::StorageMigrator::<T>::new()
+				.pre_migration()
+				.map_err(|_| "pre-migration failed")
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			migrations::StorageMigrator::<T>::new().migrate()
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			migrations::StorageMigrator::<T>::new()
+				.post_migration()
+				.map_err(|_| "post-migration failed")
+		}
+	}
 
-	/// Delegation root nodes stored on chain.
+	/// Delegation root nodes stored on chain. DEPRECATED.
```

Comment:
ah yeah we talked about that, they should go away

Reply:
How did we decide to change the deprecated types? 😅

Reply:
I think we agreed they should be moved to a deprecated file. I am unsure whether you might into problems during the migration then though.

Reply:
I think the problem is that they are still marked with `pallet::storage`, so not sure how to better structure such types.

Reply:
You should remove it from the pallet to `deprecated` and initialize it there via the `decl_storage` macro:

* PR in which a storage type was changed and moved to deprecated: https://github.com/hicommonwealth/substrate/pull/14/files#diff-00c91017af35e93cc641a7bc0d73202f7bd349f13d5bc7f0736d7a272919195dR5
* PR in which a Storage entry was moved to an associated type in the config: https://github.com/paritytech/substrate/pull/8773/files#diff-70e9723e9db62816e35f6f885b6770a8449c75a6c2733e9fa7a245fe52c4656cL621

It seems like we also need to remove the Storage Prefix: https://github.com/paritytech/substrate/pull/8773/files#diff-acaa239811fe81b5d496c1b3d5059f43de8f08cd530c4bbc44fac857612936d4R30

Reply:
Addressed in https://github.com/KILTprotocol/mashnet-node/commit/0040f2d75ca135d86b18c86cea6f54e0cb7e24e5.